frontend/src/app/App.tsx

```diff
@@ -1065,7 +1049,6 @@ export class App extends PureComponent<Props, State> {
    */
   closeDialog = (): void => {
     this.setState({ dialog: undefined })
-    this.props.hostCommunication.onModalReset()
```

Comment:
Why is this function call removed? (Ditto for `onScriptRerun`, `onScriptStop`, `onCacheClear`)

Reply:
These function calls were all removed as part of implementing a more direct approach to host messages - using `ModalReset()` / modal closure as an example:
Before, the host message would set the state variable for modal closure to true, then an if statement in `componentDidUpdate` (line 399 above) would trigger a call to `closeDialog()` which would set the dialog state in `App.tsx` and then call `onModalReset()` which just changes the `withHostCommunication` state variable for modal closure back to false.

Instead, we now have a host message for modal closure directly trigger `closeDialog()` and we no longer need a `onModalReset()` call since that "state" doesn't exist in `HostCommunicationManager`